src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageLoaderResult.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure
+{
+    public struct PageLoaderResult
```

Comment:
Should this just trigger a **full** rediscovery of all pages? What do you think is better?

I think the code you wrote is reasonable, but I don't like the approach of having two levels of caching. that can both get invalidated.